pkg/cmd/roachtest/tests/multitenant_upgrade.go

```diff
@@ -220,8 +220,21 @@ func runMultitenantUpgrade(ctx context.Context, t test.Test, c cluster.Cluster)
 		func(ctx context.Context, l *logger.Logger, rng *rand.Rand, h *mixedversion.Helper) error {
 			for _, tenant := range tenants {
 				if !tenant.running {
-					if err := startTenant(ctx, l, tenant, h.Context().FromVersion, true); err != nil {
-						return err
+					upgradingService := h.DefaultService()
+					if upgradingService.Stage == mixedversion.RunningUpgradeMigrationsStage {
```

Comment:
I think `h.DefaultService` could return `h.Tenant` in this case,
```go
func (h *Helper) DefaultService() *Service {
	if h.Tenant != nil {
		return h.Tenant
	}

	return h.System
}
```

I still have half-baked understanding here, but `h.Tenant` may not be in finalizing state.

Anyways, while I'm not sure what I said above is correct, can we use `h.IsFinalizing()`? It's shorter and it would also cover the above case,
https://github.com/cockroachdb/cockroach/blob/049c30ac6956fa8f307ba2adae8b4388f5907859/pkg/cmd/roachtest/roachtestutil/mixedversion/context.go#L242-L256

Reply:
Oh the deployment mode is `mixedversion.SystemOnlyDeployment` and we are explicitly controlling the tenants, so `h.Tenant` may indeed be `nil`. Still, `h.IsFinalizing` would be a nice nit.